pkg/resource/predicates.go

```diff
@@ -75,3 +78,47 @@ func (f *NameFilter) Generic(e event.GenericEvent) bool {
 	}
 	return false
 }
+
+// NewXRDWithClaim returns a new XRDWithClaim object.
+func NewXRDWithClaim() XRDWithClaim {
+	return XRDWithClaim{}
+}
+
+// XRDWithClaim only allows CompositeResourceDefinitions which offer a resource claim.
+type XRDWithClaim struct{}
```

Comment:
Should we use https://godoc.org/sigs.k8s.io/controller-runtime/pkg/predicate#NewPredicateFuncs here?